packages/network/lib/cors.ts

```diff
@@ -71,26 +74,75 @@ export function getDomainNameFromParsedHost (parsedHost: ParsedHost) {
   return _.compact([parsedHost.domain, parsedHost.tld]).join('.')
 }
 
-export function urlMatchesOriginPolicyProps (urlStr, props) {
-  // take a shortcut here in the case
-  // where remoteHostAndPort is null
+export function urlMatchesSuperDomainOriginProps (urlStr, props) {
+  return urlMatchesSameSiteProps(urlStr, props, true)
+}
+
+export function urlMatchesOriginProps (urlStr, props) {
   if (!props) {
     return false
   }
 
-  const parsedUrl = parseUrlIntoDomainTldPort(urlStr)
+  const parsedUrl = parseUrlIntoHostProtocolDomainTldPort(urlStr)
 
   // does the parsedUrl match the parsedHost?
   return _.isEqual(parsedUrl, props)
 }
 
-export function urlOriginsMatch (url1, url2) {
+/**
+ *
+ * @param {string} urlStr - The url string
+ * @param {ParsedHostWithProtocolAndHost|undefined} props - The broken down props from parseUrlIntoHostProtocolDomainTldPort
+ * @param {boolean} [strictPortMatch] - Whether or not to make port comparison exactly match. If set to true, this function checks to see if superDomainOriginPolicies match
+ * @returns {boolean}
+ */
+export function urlMatchesSameSiteProps (urlStr: string, props?: ParsedHostWithProtocolAndHost, strictPortMatch = false) {
```

Comment:
Personally I hate passing random bools into functions

```javascript
urlMatchesSameSiteProps(urlStr, props, true) // what does true mean?? i don't know!
```

I prefer passing in objects such that the options are named:

```javascript
urlMatchesSameSiteProps({urlStr, props, strictPortMatch: true}) //ooh, thats so much better!
```

I think typescript alleviates some of this confusion, but not in a code review.


```suggestion
export function urlMatchesSameSiteProps ({urlStr: string, props?: ParsedHostWithProtocolAndHost, strictPortMatch = false}) {
```

Reply:
I'm glad you said something because I have been considering making this an object to readability, but wasn't sure if it was overkill or not. I am also not a fan of the magic boolean